install/kubernetes/helm/istio/charts/pilot/templates/deployment.yaml

```diff
@@ -58,9 +58,9 @@ spec:
 {{- end}}
           - --domain
           - {{ .Values.global.proxy.clusterDomain }}
-{{- if .Values.global.oneNamespace }}
-          - "-a"
-          - {{ .Release.Namespace }}
+{{- if .Values.appNamespace }}
+          - --appNamespace
```

Comment:
Another option would be to load values.yaml ( which should be mounted as a config map ) and get it from there ? Injector already supports this, we can add to pilot as well.